### PR TITLE
Internal: upgrade flow to 0.145.0 & fix 23 flow suppressions

### PR DESCRIPTION
**Pull request by christianvuerings**
https://github.com/facebook/flow/releases/tag/v0.145.0

## Test Plan

* Flow tests pass in CI
* Unit tests pass

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -13,8 +13,6 @@ exact_by_default=true
 include_warnings=true
 react.runtime=automatic
 sharedmemory.hash_table_pow=21
-types_first=true
```
Comment (christianvuerings) [inline .flowconfig:16]: These settings are now the default and have been removed from flow

File: docs/src/components/PropTable.js [hunk 2]
```diff
@@ -84,7 +84,6 @@ export default function PropTable({
   const { propTableVariant, setPropTableVariant } = useAppContext();
 
   if (process.env.NODE_ENV === 'development' && Component) {
-    // $FlowIssue[prop-missing]
```
Comment (christianvuerings) [inline docs/src/components/PropTable.js:87]: `propTypes` is now specified on `Component`

File: flow-typed/npm/react-datepicker_v2.x.x.js [hunk 3]
```diff
@@ -1,8 +1,25 @@
 // react-datepicker flow-typed version based on https://github.com/Hacker0x01/react-datepicker/blob/master/docs/datepicker.md
 // LocaleData type from https://github.com/date-fns/date-fns/blob/81ab18785146405ca2ae28710cdfbb13a294ec50/src/locale/af/index.js.flow
-import React from 'react';
 
 declare module 'react-datepicker' {
+  import type { Node as ReactNode, ElementRef } from 'react';
+
+  declare type PopperPlacementPositions =
```
Comment (christianvuerings) [inline flow-typed/npm/react-datepicker_v2.x.x.js:7]: https://popper.js.org/docs/v2/constructors/#options

File: flow-typed/npm/@testing-library/react_v10.x.x.js [hunk 4]
```diff
@@ -137,7 +134,7 @@ declare module '@testing-library/react' {
   ) => Promise<HTMLElement>;
 
   declare type FindAllByRole = (
-    role: Matcher,
+    role: any,
```
Comment (christianvuerings) [inline flow-typed/npm/@testing-library/react_v10.x.x.js:137]: See https://github.com/flow-typed/flow-typed/pull/3994/files#r562751316

File: packages/gestalt/src/GroupAvatar.js [hunk 5]
```diff
@@ -181,6 +181,7 @@ export default function GroupAvatar(props: Props): Node {
       height={avatarHeight}
       position="relative"
       dangerouslySetInlineStyle={{
+        // $FlowFixMe[incompatible-type]
```
Comment (christianvuerings) [inline packages/gestalt/src/GroupAvatar.js:184]: Would require runtime changes. Better would be to change it to:

```
boxShadow: outline ? `0 0 0 2px ${colorGray0}` : undefined,
```

But then we should filter out all keys set in `__style` which have a value of `undefined`. Can be done in a follow up.

File: packages/gestalt/src/IconButton.js [hunk 6]
```diff
@@ -223,7 +223,6 @@ const IconButtonWithForwardRef: React$AbstractComponent<unionProps, unionRefs> =
   );
 });
 
-// $FlowFixMe[prop-missing] flow 0.135.0 upgrade
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButton.js:226]: Bye, bye, love to see these ones gone.

**Review by AlbertCarreras — APPROVED** [review]
Thanks! Many good changes here.

File: packages/gestalt/src/Box.js [hunk 7]
```diff
@@ -202,19 +202,7 @@ const disallowedProps = [
...
-  | Element<'div'>
-  | Element<'figcaption'>
-  | Element<'figure'>
-  | Element<'footer'>
-  | Element<'header'>
-  | Element<'main'>
-  | Element<'nav'>
-  | Element<'section'>
-  | Element<'summary'>;
+type OutputType = Element<As>;
```
Comment (dangerismycat) [inline packages/gestalt/src/Box.js:205]: oooh, nice

**Review by dangerismycat — APPROVED** [review]
Thanks Christian!!

**Pull request merged**